src/lib.rs

```diff
@@ -2528,113 +2526,38 @@ impl Build {
 
     fn apple_flags(&self, cmd: &mut Tool) -> Result<(), Error> {
         let target = self.get_target()?;
-        let arch_str = &*target.full_arch;
 
-        let arch = if target.os == "macos" {
-            match arch_str {
-                "i686" => AppleArchSpec::Device("-m32"),
```

Comment:
It seems that the `-m32`/`-m64` flag is also removed, I think it might affect correctness?

Reply:
Sorry, this was intentional, updated the issue description to match.

It won't affect correctness because:
- On Clang, the `--target` flag is enough.
- On GCC, [we already set `-m32`/`-m64`](https://github.com/rust-lang/cc-rs/blob/d2822d93afc90f9727cfe775f63701bbc793f68a/src/lib.rs#L2047-L2055), the flag was duplicated.

Reply:
I think the m32/m64 settings there does not detect arm64?

Reply:
`-m64` is wrong on Aarch64, the target does not share compiler with ARM (unlike Sparc, Mips, x86, etc.), so `-m64` isn't needed (and isn't [documented](https://gcc.gnu.org/onlinedocs/gcc/AArch64-Options.html) either).

See also https://stackoverflow.com/questions/58244095/gcc-7-error-unrecognized-command-line-option-m64.

I recall reading somewhere that it works nowadays, but only because it's deliberately ignored.

(But good that you're asking the question!)

Reply:
Thanks